gnsi/acltrie/trie.go

```diff
@@ -134,25 +156,6 @@ func (t *Trie) Insert(r *pathzpb.AuthorizationRule) error {
 
 		node = node.children[pathStr]
 	}
-	// Validate the path by ensuring that compared to all other paths with the same length and same name,
-	// all list keys do not overlap.
-	err := t.walk(func(node *trieNode, walkPath *gpb.Path) (bool, error) {
```

Comment:
Should we not error out here to avoid a bad policy from being installed?

Reply:
Since Probe prefers DENY over ALLOW, these policies are ok to install, since there's no ambiguity.

Reply:
Actually I don't see where in the README where DENY is preferred over ALLOW. If that is the case, then we're essentially saying this whole section doesn't exist, since there will never be a conflict? Since if there is overlapping policies then we will always say DENY if there is a match?

https://github.com/openconfig/gnsi/blob/main/pathz/authorization-README.md#conflict-resolution

Conflict Resolution
Policy evaluation should end with a single best match rule for the provided user / path / mode. If there is more than one 'best match', an error must be logged and evaluation should return a failure.
...
Authorization rules must be defined such that a single best match is possible. If the result of policy evaluation is more than one match, an error must be raised.

Reply:
working on readme update for this